### PR TITLE
[SPARK-20631][PYTHON][ML] LogisticRegression._checkThresholdConsistency should use values not Params

### PR DESCRIPTION
**Pull request by zero323**
## What changes were proposed in this pull request?

- Replace `getParam` calls with `getOrDefault` calls.
- Fix exception message to avoid unintended `TypeError`.
- Add unit tests

## How was this patch tested?

New unit tests.

### CODE REVIEW

**Comment by zero323** [comment]
cc @jkbradley

**Comment by yanboliang** [comment]
LGTM, merged into master and branch-2.2. Thanks!

**Comment by zero323** [comment]
Thanks @yanboliang!

File: python/pyspark/ml/tests.py [hunk 1]
```diff
@@ -807,6 +807,18 @@ def test_logistic_regression(self):
         except OSError:
             pass
 
+    def logistic_regression_check_thresholds(self):
+        self.assertIsInstance(
+            LogisticRegression(threshold=0.5, thresholds=[0.5, 0.5]),
+            LogisticRegressionModel
```
Comment (jkbradley) [inline python/pyspark/ml/tests.py]: @zero323 @yanboliang  Am I missing something?  Why should the LogisticRegression constructor return an instance of type LogisticRegressionModel?  Should you have to call fit?

**Comment by zero323** [comment]
@jkbradley It shouldn't. It is not a correct test #18085

File: python/pyspark/ml/tests.py (hunk 1 above)
Comment (yanboliang) [inline python/pyspark/ml/tests.py]: It shouldn't, but it's weird that Jenkins always passed the test. Anyway, let's fix it. Thanks for catching this.

**Pull request closed without merging**